flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -542,6 +545,16 @@ public int registerNewSubscribedShardState(KinesisStreamShardState newSubscribed
 		}
 	}
 
+	/**
+	 * Registers a metric group associated with the shard id of the provided {@link KinesisStreamShardState shardState}.
+	 */
+	private MetricGroup registerMetricGroupForShard(KinesisStreamShardState shardState) {
+		return runtimeContext
+			.getMetricGroup()
+			.addGroup("Kinesis")
+			.addGroup("shardId", shardState.getStreamShardHandle().getShard().getShardId());
```

Comment:
We probably should also add a `stream` group before `shardId`, since the Kinesis consumer allows subscribing to multiple streams.

Reply:
Great idea.